javaslang/src/main/java/javaslang/collection/HashMap.java

```diff
@@ -121,22 +121,100 @@ private HashMap(HashArrayMappedTrie<K, V> trie) {
     /**
      * Creates a HashMap of the given list of key-value pairs.
      *
-     * @param pairs A list of key-value pairs
-     * @param <K>   The key type
-     * @param <V>   The value type
+     * @param <K> The key type
+     * @param <V> The value type
      * @return A new Map containing the given entries
      */
-    @SuppressWarnings("unchecked")
-    public static <K, V> HashMap<K, V> of(Object... pairs) {
-        Objects.requireNonNull(pairs, "pairs is null");
-        if ((pairs.length & 1) != 0) {
-            throw new IllegalArgumentException("Odd length of key-value pairs list");
-        }
-        HashArrayMappedTrie<K, V> trie = HashArrayMappedTrie.empty();
-        for (int i = 0; i < pairs.length; i += 2) {
-            trie = trie.put((K) pairs[i], (V) pairs[i + 1]);
-        }
-        return wrap(trie);
+    public static <K, V> HashMap<K, V> of(K k1, V v1, K k2, V v2) {
+        return new HashMap<>(HashArrayMappedTrie.<K, V> empty().put(k1, v1).put(k2, v2));
+    }
+
+    /**
+     * Creates a HashMap of the given list of key-value pairs.
+     *
+     * @param <K> The key type
+     * @param <V> The value type
+     * @return A new Map containing the given entries
+     */
+    public static <K, V> HashMap<K, V> of(K k1, V v1, K k2, V v2, K k3, V v3) {
+        return new HashMap<>(HashArrayMappedTrie.<K, V> empty().put(k1, v1).put(k2, v2).put(k3, v3));
+    }
+
+    /**
+     * Creates a HashMap of the given list of key-value pairs.
+     *
+     * @param <K> The key type
+     * @param <V> The value type
+     * @return A new Map containing the given entries
+     */
+    public static <K, V> HashMap<K, V> of(K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4) {
+        return new HashMap<>(HashArrayMappedTrie.<K, V> empty().put(k1, v1).put(k2, v2).put(k3, v3).put(k4, v4));
```

Comment:
@ruslansennov, sorry for repeating myself, I think you could make this more compact by reusing previous factory methods, i.e.:
```java
public static <K, V> HashMap<K, V> of(K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4) {
    return of(k1, v1, k2, v2, k3, v3).put(k4, v4));
```